pkg/util/function/logger.go

```diff
@@ -0,0 +1,48 @@
+package function
+
+// Logger is used to log function.
```

Comment:
I wonder if you have seen this pattern somewhere else.

`Logger` doesn't seem like the correct name (it's not logging itself). Maybe I would call it `Wrapper` (`function.Wrapper` when you include the package name).

Reply:
Thank you very much for pointing out my mistake. More precisely, it's more like a wrapper, so wrapper is the best name.

If you think this wrapper is useful, I will try to use it to fix other possible goroutine leaks.